### PR TITLE
TDRD-217 Overwrite all values

### PR DESCRIPTION
**Pull request by thanhz**
The draft-metadata lambda can pass in properties with empty values in this case we want to delete empty values from the `fileMetadata` table but ensure we don't add empty values to the db.

Added new `propertyProtected` to `CustomMetadata` to allow us to filter on properties that shouldn't be editable i.e. UUID. Will thrown an exception if trying to add metadata that are protected.

### CODE REVIEW

**Review by Tom-Hallett — APPROVED** [review]
Looks good, thanks

File: src/main/scala/uk/gov/nationalarchives/tdr/api/service/CustomMetadataPropertiesService.scala [hunk 1]
```diff
@@ -88,7 +89,8 @@ class CustomMetadataPropertiesService(customMetadataPropertiesRepository: Custom
       metadataValues.toList,
       fp.uiordinal.getOrElse(Int.MaxValue),
       fp.allowexport,
-      fp.exportordinal.map(_.toInt)
+      fp.exportordinal.map(_.toInt),
+      Set(MandatoryMetadata, SystemProperty).contains(fp.propertygroup.get)
```
Comment (vimleshtna) [inline src/main/scala/uk/gov/nationalarchives/tdr/api/service/CustomMetadataPropertiesService.scala:93]: Can you create a variable for the protected field in class and then use it here?

`val protectedPropertyGroups = Set(MandatoryMetadata, SystemProperty)`

**Review by vimleshtna — COMMENTED** [review]
Looks good. Just a minor comment. Thanks

**Review by vimleshtna — APPROVED** [review]
Looks good. Thanks

**Pull request merged**